pilot/pkg/bootstrap/server.go

```diff
@@ -1140,6 +1140,15 @@ func (s *Server) fetchCARoot() map[string]string {
 func (s *Server) initMeshHandlers() {
 	log.Info("initializing mesh handlers")
 	// When the mesh config or networks change, do a full push.
+	s.environment.AddMeshHandler(func() {
```

Comment:
This handler is identitical to the handler function below, only with additiona lines of `spiffe.SetTrustDomain`.

Would it work if we just add that line to the mesh handler below?

Reply:
oof, thanks. note sure how this happen, that is what I meant to do